src/Microsoft.ML.Mkl.Components/VectorWhitening.cs

```diff
@@ -657,8 +657,12 @@ private void SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariableName,
                 var modelName = ctx.AddInitializer(model.Take(rank * dimension), modelDimension, "model");
                 var zeroValueName = ctx.AddInitializer((float)0);
 
-                var node = ctx.CreateNode(opType, new[] { modelName, srcVariableName, zeroValueName }, new[] { dstVariableName }, ctx.GetNodeName(opType), "");
+                var gemmOutput = ctx.AddIntermediateVariable(null, "GemmOutput", true);
```

Comment:
Can you please help me understand the rationale for this?

Reply:
The model dimensions are (r x m), where r is the user specified rank. After the gemm operation, you get an r x 1 tensor. I added a transpose to match the expected 1 x r tensor.